qupulse/hardware/awgs/tabor.py

```diff
@@ -1167,6 +1201,70 @@ def remove(self, name: str) -> None:
         self.free_program(name)
         self.cleanup()
 
+    def set_volatile_parameters(self, program_name: str, parameters: Mapping[str, ConstantParameter]):
+        """Set the values of parameters which were marked as volatile on program creation."""
+        # TODO: Add exception if program creation threw a VolatileModificationWarning, add documentation
+
+        waveform_to_segment_index, program = self._known_programs[program_name]
+        names = [name for name in [*parameters] if name in [*program._volatile_parameter_mappings]]
+        if not names:
+            print('{} has no volatile parameters with these names'.format(program_name))
+            return
+
+        # Change sequencing tables in active program and program memory
+        if program.program.depth() == 1:
+            for name in names:
+                for program_mapping in program._volatile_parameter_mappings[name]:
+                    if program_mapping[0] == 0:
+                        program.program.children[program_mapping[2]].repetition_parameter.update_constants(parameters)
+                        new_rep_count = int(program.program.children[program_mapping[2]].repetition_parameter.get_value())
+                        program.program.children[program_mapping[2]].repetition_count = new_rep_count
+                        if program_mapping[3] == 1:
+                            seg_ind = program._sequencer_tables[program_mapping[1]][program_mapping[2]][1]
+                            jump_flag = program._sequencer_tables[program_mapping[1]][program_mapping[2]][2]
+                            program._sequencer_tables[program_mapping[1]][program_mapping[2]] = (new_rep_count, seg_ind,
+                                                                                                 jump_flag)
+                            if self._current_program == program_name:
+                                self.device.send_cmd(':SEQ:SEL {}'.format(program_mapping[1] + 2))
+                                self.device.send_cmd(
```

Comment:
I would collect all changes to the current program in one place (a list or set of commands) before executing them. This makes testing a lot easier and the code possibly more performant.